### PR TITLE
LF-2850: Adding a default template for making pull requests

### PR DESCRIPTION
**Pull request by kcussen**
Implements: https://lite-farm.atlassian.net/browse/LF-2850

### CODE REVIEW

File: doc/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+## Checklist:
+
+- [ ] My code follows the style guidelines of this project
+- [ ] I have performed a self-review of my own code
+- [ ] I have commented my code, particularly in hard-to-understand areas
+- [ ] My changes generate no new warnings
+- [ ] I have added tests that prove my fix is effective or that my feature works
+- [ ] New and existing unit tests pass locally with my changes
+- [ ] Any dependent changes have been merged and published in downstream modules
+- [ ] I have checked my code and corrected any misspellings
```
Comment (Duncan-Brain) [inline doc/pull_request_template.md:42]: Consider removing  - maybe too obvious

File: doc/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+## Checklist:
+
+- [ ] My code follows the style guidelines of this project
+- [ ] I have performed a self-review of my own code
+- [ ] I have commented my code, particularly in hard-to-understand areas
+- [ ] My changes generate no new warnings
+- [ ] I have added tests that prove my fix is effective or that my feature works
+- [ ] New and existing unit tests pass locally with my changes
+- [ ] Any dependent changes have been merged and published in downstream modules
```
Comment (Duncan-Brain) [inline doc/pull_request_template.md:41]: Consider removing -- not sure this is needed

File: doc/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+**Test Configuration**:
+* Firmware version:
+* Hardware:
+* Toolchain:
+* SDK:
+
+## Checklist:
+
+- [ ] My code follows the style guidelines of this project
```
Comment (Duncan-Brain) [inline doc/pull_request_template.md:35]: Consider replacing with - I have run the precommit and/or linter

File: doc/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+- [ ] This change requires a documentation update
+
+## How Has This Been Tested?
+
+Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
+
+- [ ] Passes test case
+- [ ] Other (please explain)
+
```
Comment (Duncan-Brain) [inline doc/pull_request_template.md:24]: Consider adding - Mobile view

File: doc/pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+
+**Test Configuration**:
```
Comment (Duncan-Brain) [inline doc/pull_request_template.md:27]: Consider removing this whole section -- possibly replacing with a checkbox : there are no known specific hardware requirements. if yes specify

**Comment by Duncan-Brain** [comment]
@kcussen 

Two main things for me :
- I would personally prefer this file be in the hidden` /.github` folder
- I think the markdown headings from this template are maybe too much -- I would prefer Bolded headings to keep as much as possible "above the fold"

File: doc/pull_request_template.md (hunk 3 above)
Comment (SayakaOno) [reply doc/pull_request_template.md:35]: Doesn't precommit run automatically when doing `git commit`? Is it just for me?

Comment (Duncan-Brain) [reply doc/pull_request_template.md:35]: I do not think mine does no. Do you have any extra packages or config installed to do that?

Comment (Duncan-Brain) [reply doc/pull_request_template.md:35]: hmm I wonder if this is one of those pc/mac things 
image

Comment (Duncan-Brain) [reply doc/pull_request_template.md:35]: Yup that was it -- I guess I only installed api and webapp packages haha

Comment (Duncan-Brain) [reply doc/pull_request_template.md:35]: So @SayakaOno  would it make sense to change the recommentation to 

Consider replacing with - "The precommit and linting ran successfully"

Comment (SayakaOno) [reply doc/pull_request_template.md:35]: Sounds good! (I was just thinking we could remove it if precommit would run for everyone. Sorry, I'm too lazy...)

File: doc/pull_request_template.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+- [ ] My code follows the style guidelines of this project
+- [ ] I have performed a self-review of my own code
+- [ ] I have commented my code, particularly in hard-to-understand areas
+- [ ] My changes generate no new warnings
+- [ ] I have added tests that prove my fix is effective or that my feature works
+- [ ] New and existing unit tests pass locally with my changes
+- [ ] Any dependent changes have been merged and published in downstream modules
+- [ ] I have checked my code and corrected any misspellings
+- [ ] I have added or updated language tags for text that's part of the UI
+- [ ] I have added "MISSING" for all new language tags to languages I don't speak
```
Comment (SayakaOno) [inline doc/pull_request_template.md:44]: I would like to add "I have added a copyright to a new file". (I would also like to make the right copyright easy to access for everyone...)

File: doc/pull_request_template.md (hunk 4 above)
Comment (kcussen) [reply doc/pull_request_template.md:24]: To the "How has this been tested?" section, right?

Comment (Duncan-Brain) [reply doc/pull_request_template.md:26]: yes

File: doc/pull_request_template.md (hunk 6 above)
Comment (Duncan-Brain) [reply doc/pull_request_template.md:31]: Did you mean in this template specifically?

Comment (kcussen) [reply doc/pull_request_template.md]: Added that checkbox and the license below should someone want to copy / paste

Comment (Duncan-Brain) [reply doc/pull_request_template.md]: Yeah I don't think we want license text here. Definitely yes to the checkbox.

File: .github/pull_request_template.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
+**Description**
+
+Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
+
+Jira link: 
+
+**Type of change**
+
+Please delete options that are not relevant.
```
Comment (Duncan-Brain) [inline .github/pull_request_template.md:9]: @kcussen -- this line kind of applies to the whole template

**Comment by Duncan-Brain** [comment]
Looks like a few checklist items may have accidentally got deleted too: - [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

**Review by Duncan-Brain — APPROVED** [review]
LGTM!

**Review by SayakaOno — APPROVED** [review]

**Pull request merged**